src/main/resources/public/js/shopping_cart.js

```diff
@@ -0,0 +1,69 @@
+function decrementQuantity(bookId) {
+    var token = $('#_csrf').attr('content');
+    var header = $('#_csrf_header').attr('content');
+
+    $.ajax({
+        url:'/cart/decrement/'+bookId,
+        type:'POST',
+        beforeSend: function(xhr) {
+            xhr.setRequestHeader(header, token);
+        },
+        success:function(data){
+            $('.cart-increment-'+data.item.book.id).removeAttr("disabled");
+            if(data.item.quantity == 1){
+                $('.cart-decrement-'+data.item.book.id)[0].setAttribute("disabled", true);
+            }
+
+            updateItemValues(data.item);
+            updateItemCount(data.itemCount);
+            updateTotal(data.total);
+
+        },
+        error:function(error){
+            console.log(error);
+        }
+    });
+}
+
+function incrementQuantity(bookId){
+    var token = $('#_csrf').attr('content');
+    var header = $('#_csrf_header').attr('content');
+
+    $.ajax({
```

Comment:
this PR is the only one adding in AJAX calls, @zaidoon1  @MohamedZalat @Hajar1 what do you guys think of us using a combination of both ajax and thymeleaf

Reply:
I think we should not be doing it this way because if javascript is disabled then nothing will work. we have been redoing all the features that require javascript so it's best not to add new features that require it.